pkg/gardenlet/controller/shoot/shoot_control_reconcile.go

```diff
@@ -307,6 +307,16 @@ func (c *Controller) runReconcileShootFlow(o *operation.Operation, operationType
 			Fn:           flow.TaskFn(botanist.HibernateControlPlane).RetryUntilTimeout(defaultInterval, 2*time.Minute).DoIf(o.Shoot.HibernationEnabled),
 			Dependencies: flow.NewTaskIDs(initializeShootClients, deploySeedMonitoring, deploySeedLogging, deployClusterAutoscaler),
 		})
+		_ = g.Add(flow.Task{
+			Name:         "Destroying External DNS Entry",
```

Comment:
+1, a little bit confusing. Also, because suddenly the `Destroy` task is called (while the `Deploy`) task was called earlier. I see that this is the case where the shoot gets hibernated, though, maybe we can add a comment somewhere to make this clear.